via/static/scripts/video_player/hooks/test/use-side-by-side-layout-test.js

```diff
@@ -0,0 +1,149 @@
+import { mount } from 'enzyme';
```

Comment:
These tests can actually be entirely synchronous, if the `act` function is used to wrap dispatches of events and make effects run synchronously. The upside is that this will make the tests run faster (by default effects run only on the next animation frame) and make them easier to reason about.

The downside is that if you added debouncing in future you'd need to add a way to make that happen synchronously in tests too.

```diff
diff --git a/via/static/scripts/video_player/hooks/test/use-side-by-side-layout-test.js b/via/static/scripts/video_player/hooks/test/use-side-by-side-layout-test.js
index c7e5da6..834a9fd 100644
--- a/via/static/scripts/video_player/hooks/test/use-side-by-side-layout-test.js
+++ b/via/static/scripts/video_player/hooks/test/use-side-by-side-layout-test.js
@@ -1,6 +1,6 @@
 import { mount } from 'enzyme';
+import { act } from 'preact/test-utils';
 
-import { waitFor } from '../../../test-util/wait';
 import { useSideBySideLayout } from '../use-side-by-side-layout';
 import { TOOLBAR_WIDTH } from '../use-side-by-side-layout';
 
@@ -54,95 +54,78 @@ describe('useSideBySideLayout', () => {
     const event = new CustomEvent('hypothesis:layoutchange', {
       detail,
     });
-    document.body.dispatchEvent(event);
-  }
-
-  async function sideBySideOn() {
-    await waitFor(() => {
-      const attr = document.body.getAttribute('data-side-by-side');
-      return attr && attr === 'true';
-    }, 50);
+    act(() => {
+      document.body.dispatchEvent(event);
+    });
   }
 
-  async function sideBySideOff() {
-    await waitFor(() => {
-      const attr = document.body.getAttribute('data-side-by-side');
-      return attr && attr === 'false';
-    }, 50);
+  function sideBySideState() {
+    const attr = document.body.getAttribute('data-side-by-side');
+    if (attr === 'true') {
+      return true;
+    } else if (attr === 'false') {
+      return false;
+    } else {
+      return undefined;
+    }
   }
 
   describe('initial render', () => {
-    it('does not lay out side-by-side', async () => {
+    it('does not lay out side-by-side', () => {
       renderComponent();
-
-      await sideBySideOff();
+      assert.strictEqual(sideBySideState(), false);
     });
 
-    it('provides room for sidebar controls', async () => {
+    it('provides room for sidebar controls', () => {
       renderComponent();
-
-      await sideBySideOff();
-
+      assert.strictEqual(sideBySideState(), false);
       assert.equal(document.body.style.paddingRight, `${TOOLBAR_WIDTH}px`);
     });
   });
 
   describe('when sidebar opens', () => {
     [875, 876, 1024, 1268].forEach(bodyWidth => {
-      it('applies side-by-side layout when there is enough space', async () => {
+      it('applies side-by-side layout when there is enough space', () => {
         document.body.style.width = `${bodyWidth}px`;
         renderComponent();
         triggerEvent();
-
-        await sideBySideOn();
+        assert.isTrue(sideBySideState());
       });
     });
 
     [450, 600, 874].forEach(bodyWidth => {
-      it('allows sidebar to overlap content on narrow screens', async () => {
+      it('allows sidebar to overlap content on narrow screens', () => {
         document.body.style.width = `${bodyWidth}px`;
         renderComponent();
         triggerEvent();
-
-        await sideBySideOff();
+        assert.isFalse(sideBySideState());
       });
     });
-
-    it('does not apply side-by-side layout if client side-by-side layout already activated', async () => {
-      renderComponent();
-      triggerEvent();
-
-      await sideBySideOn();
-
-      // This indicates the client itself is handling side-by-side layout
-      fakeLayoutDetail.sideBySideActive = true;
-      triggerEvent();
-
-      return await sideBySideOff();
-    });
   });
 
   describe('when window is resized', () => {
-    it('recalculates side-by-side layout', async () => {
+    it('recalculates side-by-side layout', () => {
       document.body.style.width = '450px';
       renderComponent();
       triggerEvent();
 
-      await sideBySideOff();
+      assert.isFalse(sideBySideState());
 
-      document.body.style.width = '1450px';
-      window.dispatchEvent(new Event('resize'));
-      await sideBySideOn();
+      act(() => {
+        document.body.style.width = '1450px';
+        window.dispatchEvent(new Event('resize'));
+      });
+      assert.isTrue(sideBySideState());
     });
   });
 
   describe('when sidebar closes', () => {
-    it('does not provide room for sidebar at right', async () => {
+    it('does not provide room for sidebar at right', () => {
       renderComponent();
       fakeLayoutDetail.sidebarLayout.expanded = false;
       triggerEvent();
 
-      await sideBySideOff();
+      assert.isFalse(sideBySideState());
       assert.equal(document.body.style.paddingRight, `${TOOLBAR_WIDTH}px`);
     });
   });
```